src/data-types/datetime2.ts

```diff
@@ -1,14 +1,16 @@
+import { DataType, ParameterData } from '../data-type';
+
 const YEAR_ONE = new Date(2000, 0, -730118);
 const UTC_YEAR_ONE = Date.UTC(2000, 0, -730118);
 
-module.exports = {
+const DateTime2: DataType = {
```

Comment:
If you change the definition here to be this:

```suggestion
const DateTime2: DataType & { resolveScale: NonNullable<DataType['resolveScale']> } = {
```

TypeScript won't think `resolveScale` might be undefined for this type - and that'll allow you to revert the checks you added to see of `this.resolveScale` is defined or not. 👍